src/clib/ct.cpp

```diff
@@ -49,6 +53,149 @@ extern "C" {
+            for (int i = 0; i < na; i++) {
+                adj.push_back(SolutionCabinet::at(adjacent[i]));
+            }
+            auto sol = newInterface(infile, name, adj);
+            // add associated objects
+            ThermoCabinet::add(sol->thermo());
+            if (sol->kinetics()) {
+                KineticsCabinet::add(sol->kinetics());
+            }
+            if (sol->transport()) {
+                TransportCabinet::add(sol->transport());
+            }
+            return SolutionCabinet::add(sol);
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_size()
+    {
+        try {
+            return SolutionCabinet::size();
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_del(int n)
+    {
+        try {
+            if (n >= 0 && n < SolutionCabinet::size()) {
+                // remove all associated objects
+                auto sol = SolutionCabinet::at(n);
+                int index = ThermoCabinet::index(*(sol->thermo()));
+                if (index >= 0) {
+                    ThermoCabinet::del(index);
+                }
+                if (sol->kinetics()) {
+                    index = KineticsCabinet::index(*(sol->kinetics()));
+                    if (index >= 0) {
+                        KineticsCabinet::del(index);
+                    }
+                }
+                if (sol->transport()) {
+                    index = TransportCabinet::index(*(sol->transport()));
+                    if (index >= 0) {
+                        TransportCabinet::del(index);
+                    }
+                }
+            }
+            SolutionCabinet::del(n);
+            return 0;
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_reset()
+    {
+        try {
+            SolutionCabinet::reset();
+            return 0;
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_name(int n, int buflen, char* buf)
+    {
+        try {
+            string name = SolutionCabinet::item(n).name();
+            copyString(name, buf, buflen);
+            return int(name.size());
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_thermo(int n)
+    {
+        try {
+            auto sol = SolutionCabinet::at(n);
+            return ThermoCabinet::index(*(sol->thermo()));
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
+    }
+
+    int soln_kinetics(int n)
+    {
+        try {
+            auto sol = SolutionCabinet::at(n);
+            if (!sol->kinetics()) {
+                return -1;
+            }
+            return KineticsCabinet::index(*(sol->kinetics()));
+        } catch (...) {
+            return handleAllExceptions(-1, ERR);
+        }
```

Comment:
Using 
```c
return handleAllExceptions(npos, ncos);
```
as in other pre-existing instances.

Reply:
Sorry for being unclear here. Since the method returns an `int`, I think it's fine for `handleAllExceptions` to use int values rather than `npos` which is of type `size_t` (I'm sure we have other places where this is mixed up, but that's not that big of a problem).

The issue is the choice of sentinel value. The logic in the caller's code should be that if either of the values specified by `handleAllExceptions` is returned, an error occurred and the message for that error can be obtained by calling `ct_getCanteraError`. But if this method also returns `-1` (same thing as `npos` when it gets converted to an `int`) when the issue is just that the `Solution` doesn't have a `Kinetics` object (which is not considered an error), then the caller will incorrectly assume that there is an error message to be retrieved.

Reply:
Ok. Thanks for the clarification! (`clib`'s exception handling is something that I wasn't too familiar with.) So I guess, simply switching to
```C++
return handleAllExceptions(-2, ERR);
```
would do the trick? (with `ERR` being -999);